app/src/main/java/net/gsantner/markor/activity/DocumentActivity.java

```diff
@@ -173,30 +173,46 @@ protected void onNewIntent(Intent intent) {
     private void handleLaunchingIntent(final Intent intent) {
         if (intent == null) return;
 
-        String intentAction = intent.getAction();
-        Uri intentData = intent.getData();
+        final String intentAction = intent.getAction();
+        final Uri intentData = intent.getData();
 
+        // Pull the file from the intent
+        // -----------------------------------------------------------------------
         File file = (File) intent.getSerializableExtra(Document.EXTRA_PATH);
 
-        boolean intentIsView = Intent.ACTION_VIEW.equals(intentAction);
-        boolean intentIsSend = Intent.ACTION_SEND.equals(intentAction);
-        boolean intentIsEdit = Intent.ACTION_EDIT.equals(intentAction);
-        boolean showedShareInto = false;
+        final boolean intentIsView = Intent.ACTION_VIEW.equals(intentAction);
+        final boolean intentIsSend = Intent.ACTION_SEND.equals(intentAction);
+        final boolean intentIsEdit = Intent.ACTION_EDIT.equals(intentAction);
 
         if (intentIsSend && intent.hasExtra(Intent.EXTRA_TEXT)) {
-            showedShareInto = showShareInto(intent);
+            showShareInto(intent);
+            return;
         } else if (Intent.ACTION_PROCESS_TEXT.equals(intentAction) && intent.hasExtra(Intent.EXTRA_PROCESS_TEXT)) {
             intent.putExtra(Intent.EXTRA_TEXT, intent.getStringExtra("android.intent.extra.PROCESS_TEXT"));
-            showedShareInto = showShareInto(intent);
+            showShareInto(intent);
+            return;
         } else if (file == null && (intentIsView || intentIsEdit || intentIsSend)) {
             file = _cu.extractFileFromIntent(this, intent);
+            if (file == null) {
+                // More permissive - file may not exist
+                // Will be filtered out in next stage
+                file = MarkorContextUtils.getIntentFile(intent, null);
+            }
         }
 
-        if (file != null && (file.isDirectory() || !FormatRegistry.isFileSupported(file))) {
+        // Decide what to do with the file
+        // -----------------------------------------------------------------------
+
+        if (file == null) {
+            final String msg = getString(R.string.filemanager_doesnot_supply_required_data__appspecific) + "\n\n" + getString(R.string.sync_to_local_folder_notice);
+            showErrorMessage(Html.fromHtml(msg.replace("\n", "<br/>")));
+        } else if (file.canRead() || GsFileUtils.canCreate(file)) {
```

Comment:
We warn on files which can't be read or created. If we ask to open a file which does not exist but can be created we will allow it's creation